be/src/formats/json/struct_column.cpp

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "formats/json/struct_column.h"
+
+#include "column/struct_column.h"
+#include "formats/json/nullable_column.h"
+#include "gutil/strings/substitute.h"
+
+namespace starrocks {
+
+Status add_struct_column(Column* column, const TypeDescriptor& type_desc, const std::string& name,
+                         simdjson::ondemand::value* value) {
+    auto struct_column = down_cast<StructColumn*>(column);
+
+    try {
+        simdjson::ondemand::object obj = value->get_object();
+
+        for (size_t i = 0; i < type_desc.children.size(); i++) {
+            const auto& field_name = type_desc.field_names[i];
+            const auto& field_type_desc = type_desc.children[i];
+
+            auto field_column = struct_column->field_column(field_name);
+            simdjson::ondemand::value field_value = obj.find_field_unordered(field_name);
+            RETURN_IF_ERROR(add_nullable_column(field_column.get(), field_type_desc, name, &field_value, true));
```

Comment:
name OR field_name ?
What is the `name` for?

Reply:
The `name` is the column name. It's used to report error.